internal/external-dns/plan/plan.go

```diff
@@ -292,6 +307,11 @@ func (p *Plan) Calculate() *Plan {
 			var rTypeUpdate endpointUpdate
 			recordsByType := t.resolver.ResolveRecordTypes(key, row)
 			for _, records := range recordsByType {
+				// handle soft_delete
+				if records.current != nil && records.current.Labels != nil && records.current.Labels[SoftDeleteLabel] == "true" {
+					p.processDelete(records, key, &managedChanges)
+					continue
+				}
```

Comment:
Hmm :thinking:, i see you are actually checking for the existence of this label on the current endpoint rather than desired, why is that?